src/components/timecard/start-finish-time-input/StartFinishTimeInput.jsx

```diff
@@ -24,42 +29,80 @@ const StartFinishTimeInput = ({
 
   useEffect(() => {
     updateErrorMessages();
+    updateFinishTimeText();
   }, [formState]);
 
   const updateErrorMessages = () => {
     const findErrors =
       errors &&
-      Object.keys(errors).filter((inputName) => inputName.includes(name));
+      Object.values(errors).filter(
+        (error) =>
+          error.inputName === 'shift-start-time' ||
+          error.inputName === 'shift-finish-time'
+      );
     let relevantErrorMessages = [];
     if (findErrors) {
-      relevantErrorMessages = findErrors.map((inputName) => {
-        return errors[inputName].message;
+      relevantErrorMessages = findErrors.map((error) => {
+        return error.message;
       });
     }
     setErrorMessages(relevantErrorMessages);
   };
 
+  const updateFinishTimeText = () => {
+    const startTime = getFormValues(inputNames.shiftStartTime);
+    const finishTime = getFormValues(inputNames.shiftFinishTime);
+    const nextDay = isFinishTimeOnNextDay(startTime, finishTime);
+    setFinishTimeText(nextDay ? 'Finishes next day' : '');
+    setLocalEndDate(
+      nextDay ? dayjs(localStartDate).add(1, 'day') : dayjs(localStartDate)
+    );
+  };
+
   return (
     <div
       className={`govuk-form-group ${
-        errorMessages.length > 0 && 'govuk-form-group--error'
+        errorMessages.length > 0 &&
+        Object.values(errors).some(
+          (error) =>
+            error.inputName === 'shift-start-time' ||
+            error.inputName === 'shift-end-time'
+        )
+          ? 'govuk-form-group--error'
+          : ''
       }`}
     >
-      <div className="govuk-grid-row" data-testid="error-box">
-        {sortErrorKeys(errors, desiredErrorOrder).map((error, i) => (
+      {/* kinda works, need to fix the error messages so the correct one is displayed */}
+      {/* <div className="govuk-grid-row" data-testid="error-box">
+        {errorMessages.map((message, i) => (
           <div
             id={`${name}-${i}-error`}
             key={i}
             className="govuk-error-message govuk-!-margin-left-3"
           >
-            {errors[error]?.message && (
-              <div>
-                <span className="govuk-visually-hidden">Error:</span>
-                {errors[error]?.message}
-              </div>
-            )}
+            <span className="govuk-visually-hidden">Error:</span>
+            {message}
           </div>
         ))}
+      </div> */}
+
+      <div className="govuk-grid-row" data-testid="error-box">
+        {sortErrors(errors).map((error) => {
```

Comment:
Sorry, Laurie. Is that code dependent on another piece of work from Ian or Xeena? Tried to run it locally and seems like I am getting errors here. Specifically on line 90` {sortErrors(errors).map((error) => {` 
I am trying to add time for a new day. I am selecting Shift as a time period. Clicking Continue btn and then the console is throwing an error. 
I guess, it is because sortErrors(errors) is assuming that 'errors' is not empty/null and it is maybe trying to sort errors ?  
I checked the sortErrors.js file and sortErrors(errors) is jumping straight away to .forEach(errors) without checking if the var is empty/null. From what I understand, errors should be an array of objects. Maybe it needs something simple like null or array length validation? 
_That is my guess. It can be totally wrong. I guess you have a better picture of it, so I am happy to arrange a quick chat if you don't mind._